buffer.go

```diff
@@ -353,7 +380,27 @@ func (p *bufferPool) put(b *buffer) {
 	if b.pool != p {
 		panic("BUG: buffer returned to a different pool than the one it was allocated from")
 	}
-	p.pool.Put(b)
+	// if this slice is somehow less then our min pool size, just drop it
+	sz := cap(b.data)
+	if sz < basePoolIncrement {
+		return
+	}
+	i := levelledPoolIndex(sz / 2) // divide by 2 to put the buffer in the level below so it will always be large enough
```

Comment:
In cases where the program did not grow the byte slice with `append`, would this division result in always returning the buffer to the `sync.Pool` _below_ the one it was allocated from?

Reply:
It's my intention for it to go back into the same pool.

If, for instance, you request a 2500 byte buffer it would belong to pool index 2.
pool 0: 0 -> 1023
pool 1: 1024 -> 2047
pool 2: 2048 -> 4095

these lines would align it to the beginning of the next pool. so the code actually allocates a 4096 byte buffer.
```
		poolSize := basePoolIncrement << i
		if poolSize > sz {
			sz = poolSize
		}
```

then on put 4096 / 2 = 2048 which goes back in the pool index 2.